### DISCUSSION TITLE
Parse Time without minute and set it to 00 as default

### DISCUSSION THREAD

**Discussion by Zercerium** [time-rs/time#590] [Other]
Hi,
I know it is possible to parse time without seconds, and they will be set to 0. I think it would also be convincing to do the same just with minutes or?

```rust
fn test_time() {
        let time_str = "15";
        let format = format_description!("[hour]");
        let datetime = Time::parse(time_str, &format).unwrap();
        println!("{}", datetime);
}
```

**Comment by jhpratt** [comment]
Is there any situation where a format such as this is used on a regular basis?

**Comment by Zercerium** [comment]
In some logs and equivalent things were only every hour things are collected/aggregated.
For example i deal with the following format: YYYYMMDDhh

At the moment I append 00 to the strings so that I don't have to parse Date, create Time, set hour, merge date and time together.

**Reply by jhpratt** [reply]
When I say on a regular basis, I mean it would be relatively common to run across. I'm not asking if you specifically had a use case — I figured you did given that you asked.

**Reply by Zercerium** [reply]
sorry for the misunderstanding, I cannot provide really common sources where this format is used. I just thought this could be convenient without a drawback because you specify the format by yourself.

**Reply by jhpratt** [reply]
The drawback is that it would accept values where a user is plausibly missing information that should be present.

**Reply by Zercerium** [reply]
i my opinion it should be visible that there are no minutes if you omit it in the format `format_description!("[hour]");` like with seconds. It should also error if the string is not completely parsed and there are numbers left in the provided string.
I would close it now since answered.
Thanks for your time.
